site/content/entry/yuzu-progress-report-sep-2023/index.md

```diff
@@ -0,0 +1,379 @@
+  >}}
+
+[GPUCode](https://github.com/GPUCode) joined the fray too this month with several code additions.
+
+First of all, something *light*, let’s introduce {{< gh-hovercard "11535" "separate command buffer uploads!" >}}
+This gives the benefit of a slight performance uplift, caused by reduced renderpass breaks, which greatly helps mobile GPUs. 
+The larger the renderpass, the longer mobile GPUs can keep data in tile memory, so breaking it is a costly operation. Dedicated GPUs benefit from this change, but much less.
+
+We measured an 8% performance increase in some games with a 5800X3D. 
+Lower end systems could see bigger gains.
+
+Round 1. 
+GPUCode also fought against the rendering issues affecting `Mortal Kombat 1` and `Sonic Forces` when using Vulkan.
+For those deep into the yuzu lore, this is a compute shader implementation of {{< gh-hovercard "11556" "MSAA image copies," >}} similar to what epicboy [did back in February](https://yuzu-emu.org/entry/yuzu-progress-report-feb-2023/#other-gpu-and-video-changes).
+As said back then, compared to OpenGL, Vulkan requires a lot more work.
+
+{{< single-title-imgs-compare
+	"Emutality! (Mortal Kombat 1)"
+	"./mk1bug.png"
+	"./mk1fix.png"
+>}}
+
+This change also improves the original shader, allowing games like `Pinball FX3` to render and work correctly now, regardless of the graphics API in use, while also improving the rendering of games like `Fate/EXTELLA: The Umbral Star`.
+
+{{< imgs
+	"./msaa1.png| I ask of you. Are you my Master? (Fate/EXTELLA: The Umbral Star)"
+  >}}
+
+{{< imgs
+	"./msaa2.png| Don’t break it, please (Pinball FX3)"
+  >}}
+
+Players of `The Legend of Nayuta: Boundless Trails`, and other Falcom games like `The Legend of Heroes` series, reported that text was rendered incorrectly, displaying weird colour highlights.
+GPUCode investigated this and found that we were using the wrong colour component order (BGR vs RGB for example) for the `A4B4G4R4_UNORM` texture format.
+Instead, `A4B4G4R4_UNORM_PACK16_EXT` has {{< gh-hovercard "11557" "the order" >}} the game expects, providing correct rendering.
+
+{{< single-title-imgs-compare
+	"Ask Jobs what he thinks about good font rendering (The Legend of Nayuta: Boundless Trails)"
+	"./nayubug.png"
+	"./nayufix.png"
+>}}
+
+Continuing this trend of texture fixes, GPUCode also identified a bug in how sRGB textures were being handled.
+
+The problem was that sRGB textures could not be used as storage images in shaders. 
+The solution was to create the texture as linear, which is another colour space that does not have gamma correction, and then use an sRGB view to access it. 
+This way, the texture can be used as a storage image and still have the correct colours.
+
+However, this solution had an unintended side effect. 
+When the texture was used directly without the view–for example, when it was resolved by copying it to another image–the colours were wrong because the colour space had changed. 
```

Comment:
```suggestion
When the texture was used directly without the view — for example, when it was resolved by copying it to another image — the colours were wrong because the colour space had changed. 
```